### PR TITLE
Fail the SplitUpdate if table's any child has update triggers

### PR DESCRIPTION
**Pull request by adam8157**
Check `make_splitupdate_path()`, it is supposed to fail, but Greenplum
skipped the check on the table's children while merging PostgreSQL 12.

This commit gets the check back.

### CODE REVIEW

**Comment by adam8157** [comment]
Pipeline: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-split_update_triggers-rocky8

File: src/backend/utils/cache/lsyscache.c [hunk 1]
```diff
@@ -4559,63 +4576,3 @@ child_distribution_mismatch(Relation rel)
...
 
-/*
- *  child_triggers
- *  Return true if the table is partitioned and any of the child partitions
- *  have a trigger of the given type.
- */
-bool
-child_triggers(Oid relationId, int32 triggerType)
-{
-/* GPDB_12_MERGE_FIXME */
```
Comment (zxuejing) [inline src/backend/utils/cache/lsyscache.c:4570]: Hi @adam8157, In inheritance_planner we have already processed children of partition table
and create path for each child.
So each child will call has_update_triggers. And we just check whether relation of relid
has update trigger, do not check its children.

Comment (adam8157) [reply src/backend/utils/cache/lsyscache.c:4570]: Ah, correct, Postgres planner and ORCA have different behaviors, I have amended a trick patch.

File: src/test/regress/expected/triggers_gp.out [hunk 2]
```diff
@@ -75,15 +75,22 @@ create table parted_trig (partkey int, nonkey int, distkey int)
...
-create trigger trig_upd_after after update on parted_trig
-  for each row execute procedure update_notice_trig();
 create trigger trig_del_after after delete on parted_trig
   for each row execute procedure delete_notice_trig();
 -- Inserts. Should fire the INSERT trigger.
 insert into parted_trig values (1, 1, 1);
 NOTICE:  insert trigger fired on parted_trig1 for INSERT  (seg1 127.0.0.1:40001 pid=10560)
 insert into parted_trig values (2, 2, 2);
 NOTICE:  insert trigger fired on parted_trig2 for INSERT  (seg0 127.0.0.1:40000 pid=10559)
+-- Have an UPDATE trigger on a child.
```
Comment (zxuejing) [inline src/test/regress/expected/triggers_gp.out:85]: Does optimizer and planner use the same triggers_gp.out?

Comment (adam8157) [reply src/test/regress/expected/triggers_gp.out:85]: Correct, no differences.

File: src/backend/utils/cache/lsyscache.c [hunk 3]
```diff
@@ -1591,15 +1591,22 @@ get_oprjoin(Oid opno)
...
 bool
-has_update_triggers(Oid relid)
+has_update_triggers(Oid relid, bool including_children)
 {
 	Relation	relation;
 	bool		result = false;
 
-	/* Assume the caller already holds a suitable lock. */
-	relation = table_open(relid, NoLock);
+	relation = RelationIdGetRelation(relid);
```
[the remark this answers is not in the document]
Comment (adam8157) [reply src/backend/utils/cache/lsyscache.c:1609]: They are almost the same, except `table_open(..., NoLock)` asserts the caller already holds a lock, which is not true for the child tables with Postgres's partitioning implementation.

**Review by zxuejing — APPROVED** [review]

**Pull request merged**